Gem/Code/Source/Components/NetworkPlayerMovementComponent.cpp

```diff
@@ -132,16 +170,55 @@ namespace MultiplayerSample
         GetEntity()->GetTransform()->SetLocalRotationQuaternion(newOrientation);
 
         // Update velocity
-        UpdateVelocity(*playerInput);
+        UpdateVelocity(*playerInput, deltaTime);
 
-        GetNetworkCharacterComponentController()->TryMoveWithVelocity(GetVelocity(), deltaTime);
+        // absolute velocity is based on velocity generated by the player and other sources
+        const AZ::Vector3 absoluteVelocity = GetBaseVelocity() + GetLocalVelocity();
+
+        // if we're not moving down on a platform and have a negative velocity we're falling
+        GetNetworkAnimationComponentController()->ModifyActiveAnimStates().SetBit(
+            aznumeric_cast<uint32_t>(CharacterAnimState::Falling), !onGround && absoluteVelocity.GetZ() < 0.f);
```

Comment:
this is where the ramp issue comes in, I think I need to adjust the players foot/ground contact point a bit

Reply:
Don't be afraid to reach out to our animation folks as well.. the interaction between networking, physics, and animation is incredibly subtle when it comes to player controlled characters, especially if we want things to feel great.

Reply:
When we apply forward velocity while a player moves on a ramp we do not apply it in the slope of the ramp so if you are walking fast enough you will exceed the force of gravity keeping you on the ramp temporarily and your ground contact will say you're not on the ground and you get that fun stair-stepping movement.

It'd be better to have the player slide along surfaces instead but currently it looks like we only support "sliding" up a ramp or stairs based on the slope height, it doesn't try to "stick" you to ramps or stairs going down

Reply:
I went ahead and added a simple slope detection mechanism to make the player stick when going down ramps.  Doesn't work so well on stairs yet but probably could with some tweaking.

Reply:
Yeah, there's also a nuance in PhysX which I haven't figured out a particularly good way to deal with, where if you're moving slowly enough the logic for dealing with steps will allow you to climb a slope which is steeper than the maximum slope you're supposed to be able to deal with.